graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -268,47 +314,178 @@ def enter(self, node, key, parent, path, ancestors):
+    def __init__(self, renamings: Dict[str, Optional[str]], query_type: str) -> None:
+        """Create a visitor to check that suppression does not cause an illegal state.
+
+        Args:
+            renamings: Dict[str, Optional[str]], from original field name to renamed field name. Any
+                       name not in the dict will be unchanged
+            query_type: str, name of the query type (e.g. RootSchemaQuery)
+        """
+        # Note that as field names and type names have been confirmed to match up, any renamed
+        # query type field already has a corresponding renamed type. If no errors, due to either
+        # invalid names or name conflicts, were raised when renaming type, no errors will occur when
+        # renaming query type fields.
+        self.in_query_type = False
+        self.renamings = renamings
+        self.query_type = query_type
+
+    def enter_object_type_definition(
+        self,
+        node: ObjectTypeDefinitionNode,
+        key: Optional[Any],
+        parent: Optional[Any],
+        path: List[Any],
+        ancestors: List[Any],
+    ) -> None:
+        """If the node's name matches the query type, record that we entered the query type."""
+        if node.name.value == self.query_type:
+            self.in_query_type = True
+
+    def leave_object_type_definition(
+        self,
+        node: ObjectTypeDefinitionNode,
+        key: Optional[Any],
+        parent: Optional[Any],
+        path: List[Any],
+        ancestors: List[Any],
+    ) -> None:
+        """If the node's name matches the query type, record that we left the query type."""
+        if node.name.value == self.query_type:
+            self.in_query_type = False
+
+    def enter_field_definition(
+        self,
+        node: FieldDefinitionNode,
+        key: Optional[Any],
+        parent: Optional[Any],
+        path: List[Any],
+        ancestors: List[Any],
+    ) -> None:
+        """If not at query type, check that no field depends on a type that was suppressed."""
+        if self.in_query_type:
+            return None
+        # At a field of a type that is not the query type
+        field_name = node.name.value
+        node_type = get_ast_with_non_null_and_list_stripped(node.type)
+        if node_type == REMOVE:
+            # Then this field depends on a type that was suppressed, which is illegal
+            # Note that we refer to the last item in ancestors, which is not the same as parent. The
+            # last item in ancestors is actually the grandparent of node, specifically the
+            # ObjectTypeDefinitionNode that has this field. We use the grandparent node instead of
+            # the parent because the parent of a FieldDefinitionNode is simply a list of
+            # FieldDefinitionNodes, which doesn't contain the name of the type containing this node
+            # (which we need for the error message).
+            type_name = ancestors[-1].name.value
```

Comment:
Better than before with `*args` but still not great (see preceding comment)

Also, although it's impossible for ancestors to be `None` inside a `FieldTypeDefinitionNode` I'm going to add an assertion in code for stuff like this tomorrow